stdlib/2/itertools.pyi

```diff
@@ -35,7 +35,7 @@ def ifilterfalse(predicate: Optional[Callable[[_T], Any]],
 def groupby(iterable: Iterable[_T]) -> Iterator[Tuple[_T, Iterator[_T]]]: ...
 @overload
 def groupby(iterable: Iterable[_T],
-            key: Callable[[_T], _S]) -> Iterator[Tuple[_S, Iterator[_T]]]: ...
+            key: Optional[Callable[[_T], _S]]) -> Iterator[Tuple[_S, Iterator[_T]]]: ...
```

Comment:
Shouldn't we add a separate overload instead? If key is None, there's nothing to constrain the value of _S.

Reply:
@JelleZijlstra Good point, opened #2794.